githubstat/metrics_commit_request.go

```diff
@@ -0,0 +1,123 @@
+package githubstat
+
+import (
+	"fmt"
+	"github.com/google/go-github/github"
+	"strings"
+	"time"
+)
+
+type WarpRepositoryCommit struct {
+	RepositoryCommit *github.RepositoryCommit
+	Owner            string
+	Repo             string
+	MergedAt         *time.Time
+}
+
+// sameCommitter means the pr author was same with the commit author
+func (m *WarpRepositoryCommit) findMergedTime(client *github.Client, sameCommitter bool) bool {
+	SHA := m.RepositoryCommit.SHA
+	owner := m.Owner
+	repo := m.Repo
+	author := m.RepositoryCommit.Author.Login
+	if !sameCommitter {
+		author = m.RepositoryCommit.Committer.Login
+	}
+	// TODO consider using template
+	query := *SHA + " repo:" + owner + "/" + repo + " type:pr" + " author:" + *author
+	pr := findPRfromCommit(client, query)
+	if pr == nil {
+		return false
+	}
+	m.MergedAt = pr.ClosedAt
+	return true
+}
+
+func listCommits(client *github.Client, owner string, repo string, author string) ([]*github.RepositoryCommit, error) {
+	opt := &github.CommitsListOptions{
+		Author:      author,
+		ListOptions: github.ListOptions{PerPage: 100, Page: 1},
+		//Head:        client.userName + ":",
+	}
+
+	page := 1
+	var allCommits []*github.RepositoryCommit
+	for {
+		commits, resp, err := client.Repositories.ListCommits(owner, repo, opt)
```

Comment:
it would be better if we consider filter out commits by `CommitsListOptions.Since` and `CommitsListOptions.Until`. for example, commits committed after `CommitsListOptions.Until` is absoultely merged after `CommitsListOptions.Until`